File: tests/workflow/producer/rust/src/producer/events/finishconsumertest.rs
use super::{identification, producer::Control, protocol, scope::AnonymousScope, Context};
use crate::stat::Alias;
use clibri::server;
use std::str::FromStr;
use uuid::Uuid;

type BroadcastFinishConsumerTestBroadcast = (Vec<Uuid>, protocol::FinishConsumerTestBroadcast);

#[allow(unused_variables)]
pub async fn emit<E: server::Error, C: server::Control<E>>(
    event: protocol::FinishConsumerTest,
    scope: &mut AnonymousScope<'_, E, C>,
) -> Result<BroadcastFinishConsumerTestBroadcast, String> {
    let uuid = match Uuid::from_str(&event.uuid) {
        Ok(uuid) => uuid,
        Err(err) => {
            return Err(format!("Fail to parse uuid {}: {:?}", event.uuid, err));
        }
    };
    scope
        .context
        .inc_stat(uuid, Alias::FinishConsumerTestBroadcast);
    Ok((vec![uuid], protocol::FinishConsumerTestBroadcast {}))
}

File: tests/workflow/producer/typescript/src/producer/events/finishconsumertest.ts
import { Filter, Producer, Context, Protocol } from "../implementation/events";
import { Output } from "../implementation/events/finishconsumertest";
import { Alias } from "../../stat";

export function emit(
	event: Protocol.FinishConsumerTest,
	filter: Filter,
	context: Context,
	producer: Producer
): Promise<Output> {
	const stat = context.getStat(event.uuid);
	stat.case(Alias.FinishConsumerTestBroadcast);
	return Promise.resolve(
		new Output()
			.broadcast([event.uuid])
			.FinishConsumerTestBroadcast(
				Protocol.FinishConsumerTestBroadcast.defaults()
			)
	);
}
